src/basic/fileio.c

```diff
@@ -780,6 +781,7 @@ static int merge_env_file_push(
 
         if (!env_name_is_valid(key)) {
                 log_error("%s:%u: invalid variable name \"%s\", ignoring.", strna(filename), line, key);
+                free(value);
```

Comment:
@keszybz , ok

Unrelated to this PR. I've added BOM
```diff
diff --git a/src/test/test-fileio.c b/src/test/test-fileio.c
index b1d688c..e8a2410 100644
--- a/src/test/test-fileio.c
+++ b/src/test/test-fileio.c
@@ -55,7 +55,7 @@ static void test_parse_env_file(void) {
         f = fdopen(fd, "w");
         assert_se(f);

-        fputs("one=BAR   \n"
+        fputs("\xef\xbb\xbfone=BAR   \n"
               "# comment\n"
               " # comment \n"
               " ; comment \n"
```
And test fails with
```
Assertion 'streq_ptr(a[0], "one=BAR")' failed at src/test/test-fileio.c:87, function test_parse_env_file(). Aborting.
```
Does it matter?